### ISSUE TITLE
Reinitialize compaction services when configuration changes

### ISSUE THREAD

**Issue by keith-turner** [apache/accumulo#1609]
The changes in #1605 create compaction services based on configuration at tablet server startup.  If the configuration in zookeeper changes after startup, the compaction services never change.  It would be nice to update the compaction services whenever configuration changes.

When the configuration for an existing compaction service goes away, instances of that service should finish any running compactions and not accept any new work while waiting.

When the configuration for existing compaction service changes, it should resize existing thread pools and/or delete thread pools that are no longer needed.

When the configuration for a compaction service that does not currently exist is noticed, then it should be created.

**Comment by keith-turner** [comment]
I have implemented this at the following branch.  

https://github.com/keith-turner/accumulo/tree/accumulo-1609

Plan to submit a PR after #1605 is merged.

**Comment by milleruntime** [comment]
The autoformat is line wrapping your change in 79611ee.  Please fix the comment.

**Comment by milleruntime** [comment]
> The autoformat is line wrapping your change in [79611ee](https://github.com/apache/accumulo/commit/79611eef97c07e3605c221812792644008e0c86a). Please fix the comment.

@keith-turner
